lib/rules/string-match.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @fileoverview Rule to flag matching strings
+ * @author Jonny Arnold
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow strings that match a given regular expression",
+            category: "Stylistic Issues",
+            recommended: false
+        },
+
+        schema: [
+            {
+                type: "string"
+            }
+        ]
+    },
+
+    create(context) {
+
+        //--------------------------------------------------------------------------
+        // Helpers
+        //--------------------------------------------------------------------------
+
+        const pattern = context.options[0] || "^$",
```

Comment:
A default pattern of `/^$/` would match empty strings. This seems like it might be confusing behavior for when no option is specified; I'm not sure why a project would want to ban using empty strings.